src/praisonai-agents/praisonaiagents/session/api.py

```
@@ -589,6 +596,46 @@ def send_message(self, message: str, **kwargs) -> str:
+    def close(self) -> None:
+        """Close and cleanup the session."""
+        if self.is_remote:
+            return  # No cleanup needed for remote sessions
+        
+        # Properly cleanup memory
+        if hasattr(self, '_memory') and self._memory:
+            try:
+                # Call close_connections to properly cleanup memory adapters
+                if hasattr(self._memory, 'close_connections'):
+                    self._memory.close_connections()
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)